sdk/eventhub/Azure.Messaging.EventHubs/src/Authorization/SharedAccessSignatureCredential.cs

```diff
@@ -84,5 +84,22 @@ public SharedAccessSignatureCredential(SharedAccessSignature signature)
         ///
         public override ValueTask<AccessToken> GetTokenAsync(TokenRequestContext requestContext,
                                                              CancellationToken cancellationToken) => new ValueTask<AccessToken>(GetToken(requestContext, cancellationToken));
+
+        internal void UpdateEventHubResourceAndValidity(string eventHubResource, TimeSpan? signatureValidityDuration)
+        {
+            SharedAccessSignature = new SharedAccessSignature(eventHubResource,
```

Comment:
This introduces some potential race conditions;  I think we'll want to lock on the sync root for these.